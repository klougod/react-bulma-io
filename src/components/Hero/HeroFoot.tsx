import React, { ReactNode } from 'react'

interface HeroFootProps {
  className?: string
  children?: ReactNode
}
export const HeroFoot = ({ className, children }: HeroFootProps) => (
  <div className={`hero-foot ${className}`}>{children}</div>
)

HeroFoot.defaultProps = {
  className: ''
}

export default HeroFoot
