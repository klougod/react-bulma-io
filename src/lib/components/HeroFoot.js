import React from 'react'
import PropTypes from 'prop-types'

export const HeroFoot = props => {

  const {
    className,
    children
  } = props

  return (
    <div className={`hero-foot ${className}`}>
      { children }
    </div>
  )
}

HeroFoot.defaultProps = {
  className: ''
}

HeroFoot.propTypes = {
  className: PropTypes.string,
  children: PropTypes.node.isRequired
}

export default HeroFoot