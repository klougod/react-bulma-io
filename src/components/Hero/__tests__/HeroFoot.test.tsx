import React from 'react'
import Enzyme from 'enzyme'

import HeroFoot from '../HeroFoot'

describe('HeroFoot component test', function () {
  it('should render without crash without props', function () {
    const heroFootMock = Enzyme.shallow(<HeroFoot />)
    expect(heroFootMock).toMatchSnapshot()
  })

  it('should render without crash with props', function () {
    const heroFootMockProps = {
      className: 'heroFoot CLass',
    }
    const heroFootMock = Enzyme.shallow(<HeroFoot {...heroFootMockProps} />)
    expect(heroFootMock).toMatchSnapshot()
  })
})
