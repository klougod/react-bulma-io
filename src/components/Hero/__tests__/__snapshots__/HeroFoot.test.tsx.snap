// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeroFoot component test should render without crash with props 1`] = `
<div
  className="hero-foot heroFoot CLass"
/>
`;

exports[`HeroFoot component test should render without crash without props 1`] = `
<div
  className="hero-foot "
/>
`;
